### PR TITLE
Clarify reference to Astro.currentLocale in /en/guides/internationalization.mdx

### PR DESCRIPTION
**Pull request by Because789**
#### Description

Proposal to improve the "Browser language detection" paragraph in /en/guides/internationalization.mdx

#### Related issues & labels (optional)

- Closes: https://github.com/withastro/docs/issues/7152
-  Helps avoiding: https://github.com/withastro/astro/issues/10017

### CODE REVIEW

File: src/content/docs/en/guides/internationalization.mdx [hunk 1]
```diff
@@ -263,16 +263,20 @@ When using functions from the [`astro:i18n` virtual module](#virtual-module-astr
...
+- [`Astro.preferredLocaleList`](/en/reference/api-reference/#astropreferredlocalelist): An array of all locales that are both requested by the browser and supported by your website. This produces a list of all compatible languages between your site and your visitor. The value is `[]` if none of the browser's requested languages are found in your `locales` array. If the browser does not specify any preferred languages, then this value will be [`i18n.locales`].
 
 In order to successfully match your visitors' preferences, provide your `codes` using the same pattern [used by the browser](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax).
 
+**Static sites** and pre-rendered pages have access to `Astro.currentLocale`.
+
+- [`Astro.currentLocale`](/en/reference/api-reference/#astrocurrentlocale): The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to [`i18n.defaultLocale`](/en/reference/configuration-reference/#i18ndefaultlocale).
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:277]:
```suggestion
- [`Astro.currentLocale`](/en/reference/api-reference/#astrocurrentlocale): The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to [`i18n.defaultLocale`](/en/reference/configuration-reference/#i18ndefaultlocale).

In order to successfully match your visitors' preferences, provide your `codes` using the same pattern [used by the browser](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Language#syntax).
```
don't need that line here now, which makes it sound like ONLY static/prerendered sites have access. So, should probably keep all the bullet points together.

File: src/content/docs/en/guides/internationalization.mdx [hunk 2]
```diff
@@ -263,16 +263,20 @@ When using functions from the [`astro:i18n` virtual module](#virtual-module-astr
...
+
+
```
Comment (sarah11918) [inline src/content/docs/en/guides/internationalization.mdx:279]:
```suggestion
```
extra blank line we can remove!

**Review by sarah11918 — COMMENTED** [review]
Hi @Because789! Thank you so much for jumping on the issue and providing these docs improvements! (Sorry it's taken a while to get back to this, as we've had some pretty big docs tasks on our plate lately, so thank you for your patience!)

I made some suggestions below! See what you think!

**Review by Because789 — COMMENTED** [review]
Hey @sarah11918, don't you worry, I'm sure there are bigger fishes to fry 😄 

I added my thoughts on your comments. This is my very first review and I don't really know what I'm doing 😬 There is no way for me to add an alternative suggestion, right?

File: src/content/docs/en/guides/internationalization.mdx [hunk 3]
```diff
@@ -263,16 +263,20 @@ When using functions from the [`astro:i18n` virtual module](#virtual-module-astr
 
 ### Browser language detection
 
-Astro's i18n routing combined with one of Astro's  [on-demand server rendering modes (`output:'server'` or `output:'hybrid'`)](/en/guides/server-side-rendering/) allow you to access two properties for browser language detection: `Astro.preferredLocale` and `Astro.preferredLocaleList`.
+Astro's i18n routing combined with one of Astro's  [on-demand server rendering modes (`output:'server'` or `output:'hybrid'`)](/en/guides/server-side-rendering/) allow you to access two properties for browser language detection: `Astro.preferredLocale` and `Astro.preferredLocaleList`.
```
[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/guides/internationalization.mdx:266]: So maybe this means reworking this a little more!

It's important to distinguish that it's not *just* setting `server` or `hybrid` mode that makes the first two properties available. It's only available on the specific on-demand rendered pages. (Both of those modes also allow you to have static pages). 

That's why the phrase showed up twice, which you're right is clunkly. So, I think that means we need to go back to the drawing board for a sentence that is accurate *and* makes sense! Let me take another shot at it.

[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/guides/internationalization.mdx:266]: Updated my suggestion. What if we remove the clutter, and then the difference between static/SSR pages stands out more!

**Comment by sarah11918** [comment]
> This is my very first review and I don't really know what I'm doing

It's going great! This is exactly how it goes. I make some reasonable changes, you commit them, then we fight GitHub's UI to discuss the things that need more work.  :joy:

I updated my suggestion -- shortened it, which almost ALWAYS makes it better -- so see what you think about that!

File: src/content/docs/en/guides/internationalization.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (Because789) [reply src/content/docs/en/guides/internationalization.mdx:266]: Yes, you are right, less is often more, it's much better! 

All that's left is, that it seems a bit inconsistent, to put `Astro.currentLocale` as code, and `Astro.preferredLocale` and `Astro.preferredLocaleList` not. I feel like putting all three as code is the way, or is there a logic behind it?

[the remark this answers is not in the document]
Comment (sarah11918) [reply src/content/docs/en/guides/internationalization.mdx:266]: D'oh! Sorry, just a copying error. I'll "code" the others now!

**Review by sarah11918 — APPROVED** [review]
Alright! I think we're good to go! Thanks for your help and patience with this one. Now you can add GitHub reviewing to your skills... so don't hesitate to jump in to other open PRs and help out. You have a great eye for consistency which is super helpful in docs! :rocket:

**Pull request merged**